rest-service/manager_rest/test/endpoints/test_filters.py

```diff
@@ -2,181 +2,242 @@
-            },
-            IS_NULL: ['m', 'o'],
-            IS_NOT_NULL: ['p', 'n']
-        }
-        filters_mapping = create_labels_filters_mapping(self.FILTER_RULES)
-        self.assertEqual(filters_mapping, expected_filters_mapping)
-
-    def test_create_filters_mapping_fails(self):
-        with self.assertRaisesRegex(BadParametersError,
-                                    '.*have the same key.*'):
-            create_labels_filters_mapping(['a=b', 'b!=c', 'a=c'])
-
-    def test_create_filters_list(self):
-        created_filters_list = get_filters_list_from_mapping(
-            create_labels_filters_mapping(LEGAL_RULES))
-        self.assertEqual(set(created_filters_list), set(LEGAL_RULES))
-
-    def test_filters_applied(self):
+
+    def test_labels_filters_applied(self):
         dep1 = self.put_deployment_with_labels(self.LABELS)
         dep2 = self.put_deployment_with_labels(self.LABELS_2)
-        self._assert_filters_applied({EQUAL: {'a': ['b']}}, {dep1.id, dep2.id})
-        self._assert_filters_applied({NOT_EQUAL: {'c': ['z']}}, {dep1.id})
-        self._assert_filters_applied({EQUAL: {'a': ['y', 'z'], 'c': ['d']}},
+        self._assert_filters_applied([('a', ['b'], 'any_of', 'label')],
+                                     {dep1.id, dep2.id})
+        self._assert_filters_applied([('c', ['z'], 'not_any_of', 'label')],
+                                     {dep1.id})
+        self._assert_filters_applied([('a', ['y', 'z'], 'any_of', 'label'),
+                                      ('c', ['d'], 'any_of', 'label')],
                                      {dep1.id})
-        self._assert_filters_applied({EQUAL: {'a': ['b']}, IS_NOT_NULL: ['e']},
+        self._assert_filters_applied([('a', ['b'], 'any_of', 'label'),
+                                      ('e', [], 'is_not_null', 'label')],
                                      {dep2.id})
-        self._assert_filters_applied({EQUAL: {'a': ['b']}, IS_NULL: ['e']},
+        self._assert_filters_applied([('a', ['b'], 'any_of', 'label'),
+                                      ('e', [], 'is_null', 'label')],
                                      {dep1.id})
-        self._assert_filters_applied({IS_NULL: ['a']}, set())
+        self._assert_filters_applied([('a', [], 'is_null', 'label')], set())
         self._assert_filters_applied(
-            {EQUAL: {'a': ['b']}, NOT_EQUAL: {'c': ['y', 'z']}}, {dep1.id})
-
-    def test_filters_functionality_fails(self):
-        err_filters = ['a null', 'a', 'a!b']
-        for err_filter in err_filters:
-            with self.assertRaisesRegex(BadLabelsFilter,
-                                        '.*not in the right format.*'):
-                create_labels_filters_mapping([err_filter])
+            [('a', ['b'], 'any_of', 'label'),
+             ('c', ['y', 'z'], 'not_any_of', 'label')], {dep1.id})
+
+    def test_filter_rule_not_dictionary_fails(self):
+        with self.assertRaisesRegex(BadFilterRule, '.*not a dictionary'):
```

Comment:
```suggestion
        with self.assertRaisesRegex(BadFilterRule, 'not a dictionary'):
```
assertRaisesRegex works in terms of re.search, not re.match, so `.*` at the start or at the end means nothing

Reply:
Thanks, changed.